src/ring_theory/ideal_operations.lean

```diff
@@ -459,6 +468,113 @@ end surjective
 
 end map_and_comap
 
+section jacobson
+variables {R : Type u} [comm_ring R]
+
+/-- The jacobson radical of I is the pullback of the
+jacobson radical of R/I under the quotient map. -/
+def jacobson (I : ideal R) : ideal R :=
+Inf { J : ideal R | I ≤ J ∧ is_maximal J }
+
+theorem jacobson_top {I : ideal R} : jacobson I = ⊤ ↔ I = ⊤ :=
+⟨λ H, classical.by_contradiction $ λ hi, let ⟨M, hm, him⟩ := exists_le_maximal I hi in
+  lt_top_iff_ne_top.1 (lt_of_le_of_lt (show jacobson I ≤ M, from Inf_le ⟨him, hm⟩) $ lt_top_iff_ne_top.2 hm.1) H,
+λ H, eq_top_iff.2 $ le_Inf $ λ J ⟨hij, hj⟩, H ▸ hij⟩
+
+theorem mem_jacobson_iff {I : ideal R} {x : R} :
+  x ∈ jacobson I ↔ ∀ y, ∃ z, x * y * z + z - 1 ∈ I :=
+⟨λ hx y, classical.by_cases
+  (assume hxy : I ⊔ span {x * y + 1} = ⊤,
+    let ⟨p, hpi, q, hq, hpq⟩ := submodule.mem_sup.1 ((eq_top_iff_one _).1 hxy) in
+    let ⟨r, hr⟩ := mem_span_singleton.1 hq in
+    ⟨r, by rw [← one_mul r, ← mul_assoc, ← add_mul, mul_one, ← hr, ← hpq, ← neg_sub, add_sub_cancel]; exact I.neg_mem hpi⟩)
+  (assume hxy : I ⊔ span {x * y + 1} ≠ ⊤,
+    let ⟨M, hm1, hm2⟩ := exists_le_maximal _ hxy in
+    suffices x ∉ M, from (this $ mem_Inf.1 hx ⟨le_trans le_sup_left hm2, hm1⟩).elim,
+    λ hxm, hm1.1 $ (eq_top_iff_one _).2 $ add_sub_cancel' (x * y) 1 ▸ M.sub_mem
+      (le_trans le_sup_right hm2 $ mem_span_singleton.2 $ dvd_refl _)
+      (M.mul_mem_right hxm)),
+λ hx, mem_Inf.2 $ λ M ⟨him, hm⟩, classical.by_contradiction $ λ hxm,
+  let ⟨y, hy⟩ := hm.exists_inv hxm, ⟨z, hz⟩ := hx (-y) in
+  hm.1 $ (eq_top_iff_one _).2 $ sub_sub_cancel (x * -y * z + z) 1 ▸ M.sub_mem
+    (by rw [← one_mul z, ← mul_assoc, ← add_mul, mul_one, mul_neg_eq_neg_mul_symm, neg_add_eq_sub, ← neg_sub,
+        neg_mul_eq_neg_mul_symm, neg_mul_eq_mul_neg, mul_comm x y]; exact M.mul_mem_right hy)
+    (him hz)⟩
+
+end jacobson
+
+section is_local
+variables {R : Type u} [comm_ring R]
+
+/-- An ideal I is lcoal iff R/I is a local ring. -/
```

Comment:
```suggestion
/-- An ideal I is local iff R/I is a local ring. -/
```
Typo

Reply:
Here also the docstring isn't defeq to the definition.

Reply:
Fixed typo